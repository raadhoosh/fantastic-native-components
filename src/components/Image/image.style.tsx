import styled from "styled-components";
import { Image as RnImage } from "react-native";

const Image = styled(RnImage)`
width: 400px;
height: 400px;
`;

export default Image;