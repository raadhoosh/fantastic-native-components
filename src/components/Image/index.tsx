import * as React from "react";
import {ImagePropsBase,ImageStyle, Image} from "react-native";

interface IProps extends ImagePropsBase {
  style?: ImageStyle | object | Array<ImageStyle>;  
}

const RnImage = (props: IProps) => {

  return (
    <Image {...props} style={props.style} />
  );
};

export default RnImage;
