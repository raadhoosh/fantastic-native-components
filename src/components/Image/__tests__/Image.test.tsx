import Image from "../index";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

const createTestProps = (props: Object) => ({
  ...props,
});

describe("Image", () => {
  describe("rendering", () => {
    let component: ShallowWrapper;
    let props: Object;
    beforeEach(() => {
      props = createTestProps({       
        source: require('../../../screen/ImagePage/img_avatar3.png'),        
      });
      component = shallow(<Image {...props} source />);
    });
    it("should render a <Image />", () => {
      expect(component).toMatchSnapshot();
    });
  });
});
