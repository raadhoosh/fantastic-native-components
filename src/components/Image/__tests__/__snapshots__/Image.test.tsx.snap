// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image rendering should render a <Image /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RnImage
    source={
      Object {
        "testUri": "../../../src/screen/ImagePage/img_avatar3.png",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "source": Object {
        "testUri": "../../../src/screen/ImagePage/img_avatar3.png",
      },
      "style": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "source": Object {
          "testUri": "../../../src/screen/ImagePage/img_avatar3.png",
        },
        "style": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
